doc/guides/west/repo-tool.rst

```diff
@@ -1,79 +1,289 @@
+
+   west init -m https://github.com/zephyrproject-rtos/zephyr --mr v1.14.0 zp
+
+would clone the upstream official zephyr repository into :file:`zp/zephyr`,
+and check out the ``v1.14.0`` release. This command creates
+:file:`zp/.west`, and set the ``manifest.path`` :ref:`configuration option
+<west-config>` to ``zephyr`` to record the location of the manifest
+repository in the workspace. The default manifest file location is used.
+
+The ``-m`` option defaults to
+``https://github.com/zephyrproject-rtos/zephyr``. The ``--mr`` option
+defaults to ``master``. The ``--mf`` option defaults to ``west.yml``. If no
+``directory`` is given, the current working directory is used.
+
+**Option 2**: to create a workspace around an existing local manifest
+repository, use:
+
+.. code-block:: none
+
+   west init -l [--mf FILE] directory
+
+This creates :file:`.west` **next to** :file:`directory` in the file
+system, and sets ``manifest.path`` to ``directory``.
+
+As above, ``--mf`` defaults to ``west.yml``.
+
+The ``west init`` command does not clone any of the projects defined in the
+manifest file, regardless of whether ``-l`` is given. To do that, use
+``west update``.
+
+.. _west-update:
+
+``west update``
+---------------
+
+This command clones and updates the projects specified in the :term:`west
+manifest` file.
+
+.. code-block:: none
+
+   west update [-h] [--stats] [-f {always,smart}] [-k] [-r] [PROJECT ...]
+
+By default, this command:
+
+#. Parses the manifest file, usually :file:`west.yml`
+#. Clones any project repositories that are not already present locally
+#. Fetches any project revisions which are not already pulled from the
+   remote
+#. Sets each project's :ref:`manifest-rev <west-manifest-rev>` branch to the
+   revision specified for that project in the manifest file
+#. Checks out each ``manifest-rev`` in local working trees, as `detached
+   HEADs <https://git-scm.com/docs/git-checkout#_detached_head>`_
+
+To operate on a subset of projects only, specify them using the ``PROJECT``
+positional arguments, which can be either project names as given in the
+manifest file, or paths to the local project clones.
+
+To force this command to fetch from project remotes even if the revisions
+appear to be available locally, either use ``--fetch always`` or set the
+``update.fetch`` :ref:`configuration option <west-config>` to ``"always"``.
+
+For safety, ``west update`` uses ``git checkout --detach`` to check out a
+detached ``HEAD`` at the manifest revision for each updated project,
+leaving behind any branches which were already checked out. This is
+typically a safe operation that will not modify any of your local branches.
+
+If you would rather rebase any locally checked out branches instead, use
+the ``-r`` (``--rebase``) option.
+
+If you would like ``west update`` to keep local branches checked out as
+long as they point to commits that are descendants of the new
+``manifest-rev``, use the ``-k`` (``--keep-descendants``) option.
```

Comment:
I never used this `update -k` option and only now I'm realizing that a simple `git rebase` can also achieve the exact same effect because `git rebase` is in such a case smart enough to do absolutely nothing. Demonstration:
```shell
$ git status; git rev-parse HEAD
  "Your branch is ahead of 'origin/master' by 1 commit."
  2f47b177d12
# Go back in time
$ git update-ref refs/remotes/origin/master origin/master~2
$ git status
  "Your branch is ahead of 'origin/master' by 3 commits."
# Go forward in time
$ git fetch
$ git rebase
  "Current branch master is up to date."
$ git status; git rev-parse HEAD
  "Your branch is ahead of 'origin/master' by 1 commit."
  2f47b177d12
```
Now I understand this does not mean `west update --rebase` is _globally_ equivalent to `west update --keep-descendants`, however it means the difference between the two is actually quite subtle and could use an example or two (not necessarily in this page)

cc: @kkasperczyk-no

Reply:
> Now I understand this does not mean west update --rebase is globally equivalent to west update --keep-descendants, however it means the difference between the two is actually quite subtle and could use an example or two (not necessarily in this page)

Submitted at: https://github.com/zephyrproject-rtos/zephyr/pull/31201 (could not add @kkasperczyk-no as a reviewer, not enough github power?)

Reply:
I'll consider this addressed by your PR. Thanks for sending that.